moped-toolbox/populate_geometry_transform/transpose_geometry.py

```diff
@@ -0,0 +1,281 @@
+        return arg[1:]
+    else:
+        return arg
+
+def execute(sql, values, get_result=False):
+    try:
+        update = pg.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+        update.execute(
+            sql, values,
+        )
+        if get_result:
+            result = update.fetchone()
+    except psycopg2.errors.InternalError_ as e:
+        print("\n\b🛎Postgres error: " + str(e))
+        print(geojson.dumps(feature, indent=2))
+        pg.rollback()
+    else:  # no exception
+        pg.commit()
+    if get_result:
+        return result
+    else:
+        return True
+
+
+def moped_proj_features():
+    # this should make you pass an argument because it's destructive
+    sql = """
+    truncate component_feature_map;
+    truncate feature_drawn_lines;
+    truncate feature_drawn_points;
+    truncate feature_intersections;
+    truncate feature_signals;
+    truncate feature_street_segments;
+    truncate features;
+    """
+    execute(sql, [], get_result=False)
+
+
+    sql = """
+    select
+      moped_proj_components.project_id,
+      moped_components.component_id as component_archtype_id,
+      moped_proj_components.project_component_id,
+      moped_components.component_name,
+      moped_components.component_subtype,
+      feature_layers.internal_table,
+      feature_layers.id as feature_layer_id,
+      feature_id,
+      feature::character varying as feature,
+      moped_proj_features.feature_id
+    from moped_proj_features
+    join moped_proj_components on (moped_proj_components.project_component_id = moped_proj_features.project_component_id)
+    join moped_components on (moped_proj_components.component_id = moped_components.component_id)
+    left join feature_layers on (feature_layers.id = moped_components.feature_layer_id)
+    where true
+    --and feature_layers.internal_table in ('feature_signals')
+    --and feature_layers.internal_table in ('feature_street_segments')
+    --and moped_proj_components.project_id = 228
+    order by moped_proj_features.feature_id asc
+    """
+
+    cursor = pg.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+    cursor.execute(sql)
+    features = cursor.fetchall()
+
+    for record in features:
+        feature = geojson.loads(record["feature"])
+
+        if not feature.is_valid:
+            raise Exception("Invalid feature")
+
+        if True:
+            print("Project ID: ", record["project_id"])
+            print("Component Name: ", record["component_name"])
+            print("Component Subtype: ", record["component_subtype"])
+            print("Feature ID: ", record["feature_id"])
+            print("Default internal table: ", record["internal_table"])
+            print(str(record["feature_id"]) + ": " + str(feature["geometry"]["type"]))
+            pp.pprint(feature["properties"])
+            pp.pprint(feature["geometry"])
+
+        feature_id = None
+        if (record["component_name"] == "Project Extent - Generic" and
```

Comment:
i'd like to change that :/

Reply:
Me too! I think it is a change that is outside the scope of this issue, and may be a confusing tag-along, but I also think the "migration" to do it would be super quick. A snackoo if there ever were one. 🍫